### PR TITLE
improved waveformwidgetabstract class hierarchy

### PR DESCRIPTION
**Pull request by m0dB**
Improving the class hierarchy below waveformwidgetabstract, as discussed here: https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/class.20hierarchy.20refactoring (includes before and after diagrams)

### CODE REVIEW

**Comment by m0dB** [comment]
ping @daschuer :-)

File: src/waveform/widgets/nonglwaveformwidgetabstract.h [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+#include <QWidget>
+
+#include "waveform/widgets/waveformwidgetabstract.h"
+
+QT_FORWARD_DECLARE_CLASS(QString)
+
+/// NonGLWaveformWidgetAbstract is a WaveformWidgetAbstract & QWidget
```
Comment (daschuer) [inline src/waveform/widgets/nonglwaveformwidgetabstract.h:9]: This comment does not add any value. You may explain the design decision or remove the comment at all.

**Review by daschuer — APPROVED** [review]
LGTM, everything is working like before. Thank you. 
I have just left a nit pick about the comment.

File: src/waveform/widgets/nonglwaveformwidgetabstract.h (hunk 1 above)
Comment (ywwg) [reply src/waveform/widgets/nonglwaveformwidgetabstract.h:9]: This comment is in keeping with the style of the other comment, so I think it's ok to keep it.  What this is lacking is more explanation, for instance what paint method it uses instead

**Review by ywwg — APPROVED** [review]
lgtm from me as well, @daschuer do you want to merge?

**Comment by m0dB** [comment]
thx!

**Pull request merged**